data/api/subgraph.py

```diff
@@ -0,0 +1,60 @@
+import requests
+import diskcache
+import json
+import logging
+import os
+from statistics import mean
+from dotenv import load_dotenv
+from collections import defaultdict
+
+from model.constants import epochs_per_day
+
+load_dotenv()
+cache = diskcache.Cache(".beaconchain_api.cache")
+
+
+@cache.memoize(expire=(24 * 60 * 60))  # cached for 24 hours
+def get_6_month_eth_deposit_data():
+    SUBGRAPH_API_KEY = os.getenv("SUBGRAPH_API_KEY")
+    if SUBGRAPH_API_KEY:
+        API_URI = (
+            "https://gateway.thegraph.com/api/"
+            + SUBGRAPH_API_KEY
+            + "/subgraphs/id/0x540b14e4bd871cfe59e48d19254328b5ff11d820-0"
+        )
+        GRAPH_QUERY = """
+        {
+        dailyDeposits(first: 180) {
+            id
+            dailyAmountDeposited
+            }
+        }
+        """
+        try:
+            JSON = {"query": GRAPH_QUERY}
+            r = requests.post(API_URI, json=JSON)
+            return r.json()["data"]
```

Comment:
If it's possible that "data" doesn't exist, might be safer doing `r.json().get("data", {})`